File: app/src/main/java/com/fivesysdev/standwithukraine/ui/EmptyStatisticDataObserver.java
package com.fivesysdev.standwithukraine.ui;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

public class EmptyStatisticDataObserver extends RecyclerView.AdapterDataObserver {

    private final View emptyView;
    private final RecyclerView recyclerView;

    public EmptyStatisticDataObserver(RecyclerView recyclerView, View view) {
        this.emptyView = view;
        this.recyclerView = recyclerView;
        checkIfEmpty();
    }

    private void checkIfEmpty() {
        if (emptyView != null && recyclerView.getAdapter() != null) {
            boolean emptyViewVisible = recyclerView.getAdapter().getItemCount() == 0;
            if (emptyViewVisible) {
                emptyView.setVisibility(View.VISIBLE);
                recyclerView.setVisibility(View.GONE);
            } else {
                emptyView.setVisibility(View.GONE);
                recyclerView.setVisibility(View.VISIBLE);
            }
        }
    }

    @Override
    public void onChanged() {
        super.onChanged();
        checkIfEmpty();
    }

}

File: app/src/main/java/com/fivesysdev/standwithukraine/ui/EmptyStatisticDataObserver.kt
package com.fivesysdev.standwithukraine.ui

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.AdapterDataObserver
import com.fivesysdev.standwithukraine.data.ext.setVisibility

class EmptyStatisticDataObserver(
    private val recyclerView: RecyclerView, private val emptyView: View?
) : AdapterDataObserver() {
    init {
        checkIfEmpty()
    }

    private fun checkIfEmpty() {
        if (emptyView != null && recyclerView.adapter != null) {
            val emptyViewVisible = recyclerView.adapter?.itemCount == 0
            emptyView.setVisibility(emptyViewVisible)
            recyclerView.setVisibility(!emptyViewVisible)
        }
    }

    override fun onItemRangeChanged(positionStart: Int, itemCount: Int) {
        super.onItemRangeChanged(positionStart, itemCount)
        checkIfEmpty()
    }

}
